lib/spack/spack/test/conftest.py

```diff
@@ -2237,14 +2237,10 @@ def pytest_runtest_setup(item):
         pytest.skip(*only_windows_marker.args)
 
 
-def _sequential_executor(*args, **kwargs):
-    return spack.util.parallel.SequentialExecutor()
-
-
 @pytest.fixture(autouse=True)
-def disable_parallel_buildcache_push(monkeypatch):
+def disable_parallelism(monkeypatch):
```

Comment:
I don't like the idea that we aren't testing parallel execution for the things we run in parallel anywhere in our test suite.

My first thought for a fix is to tag a few tests that exercise the parallel logic and have a separate CI task that runs those tests (without this fixture and without xdist).

Can you think of something better than that?